modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/consensus/ValidationErrorStorage.scala

```diff
@@ -0,0 +1,50 @@
+package io.constellationnetwork.node.shared.infrastructure.consensus
+
+import java.time.Instant
+
+import cats.data.NonEmptyList
+import cats.effect.kernel.{Async, Clock, Ref}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.auto.autoUnwrap
+import eu.timepit.refined.types.all.PosInt
+
+trait ValidationErrorStorage[F[_], Artifact, ValidationError] {
+  def addValidationError(artifact: Artifact, errors: ValidationError): F[Unit]
```

Comment:
is `ValidationError` a single error or multiple errors? Type says one thing but property name the other

Reply:
`ValidationError` can be whatever the client chooses.